lib/steps.js

```diff
@@ -373,35 +375,122 @@ When(/I tap the (button|link) "([^"]+)"/, async (elemType, phrase) => {  // √
+    await to_manipulate.select(value);
+    await scope.afterStep(scope, {waitForShowIf: true});
+
+  } else if ( tagName === `INPUT` || tagName === `TEXTAREA` ) {
+    // Set text field values (includes any other input fields)
+    await to_manipulate.evaluate( el => { el.value = '' });
+    await to_manipulate.type( value );
+    // TODO: Add hideif to this too
+    await scope.afterStep(scope, {waitForShowIf: true});
+  }
+});
+
+When(/I tap the ?(?:choice|checkbox|radio(?: button)?)? var(?:iable)? "([^"]+)"(?: with the value "([^"]+)")?/, async (var_name, value) => {
+  /* Tap a checkbox or radio button. We need to be clear that we toggle
+  *    them, not set them to a value. (radio can be treated like a
+  *    checkbox, but not the other way around)
+  * 'value' is optional for yesno noyes checkboxes.
+  * 
+  * Examples:
+  * 1. (radio yesno) I tap the var "user.likes_fruit" with the value "True"
+  * 1. (radio yesno) I tap the var "user.likes_fruit" with the value "None"
+  * 1. (checkbox yesno) I tap the var "user.likes_fruit"
+  * 1. (radio or checkbox) I tap the var "user.fruit" with the value "apple"
+  * 
+  * Also accepted first lines:
+  * 1. I tap the checkbox variable "fruits" with the value "apple"
+  * 1. I tap the radio var "fruits" with the value "apple"
+  * 1. I tap the radio button variable "fruits" with the value "apple"
+  * 1. I tap the choice var "fruits" with the value "apple"
+  * 
+  * @param {string} var_name - Name of variable set in the YAML.
+  * @param {string} [value] - Optional. Value given to some a choice.
+  * 
+  * TODO: Sentence ideas? This kept name and value order consistent with other var step.
+  */
+  // All the possible variations of attribute names
+  let base64_var_name = await scope.base64( scope, var_name );
+  let base64_first_id = base64_var_name + '_0';
+  let { name_attr_B, name_attr_R } = await scope.checkboxNameAttribute( scope, var_name, value );
+
+  // radio: `name` = base64 var name, `value` = choice value
+  // `yesno`/`noyes` checkbox: `name` = base64 var name, value always 'True'
+  // checkbox: `name` = base64 var name + stuff + base64 choice value + stuff (see function)
+  let to_tap = await Promise.race([
+    scope.page.waitForSelector(`input[name="${base64_var_name}"][value="${ value }"] ~ label`),  // radio
+    // scope.page.waitForSelector(`*[data-saveas="${base64_var_name}"] label`),  // radio showif?
+    scope.page.waitForSelector(`input[name="${base64_var_name}"][value="True"] ~ label`),  // yesno/noyes checkbox
+    // scope.page.waitForSelector(`*[data-saveas="${base64_var_name}"] label`),  // yesno/noyes checkbox showif?
+    scope.page.waitForXPath(`//input[@id="${base64_first_id}"]/ancestor::fieldset//*[contains(@class,"danota-checkbox")]/following-sibling::label`),  // 'none of the above' checkbox
+    // 'none of the above' checkbox showif???
+    scope.page.waitForSelector(`input[name="${name_attr_B}"] ~ label`),  // checkbox
```

Comment:
pointer to docassemble code would be nice, so people know the B/R distinction exists.